survey_mailer/survey_mailer.go

```diff
@@ -0,0 +1,204 @@
+package survey_mailer
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	"github.com/dxe/adb/config"
+	"github.com/dxe/adb/model"
+	"github.com/jmoiron/sqlx"
+	"github.com/sourcegraph/go-ses"
+)
+
+func sendMissingEmail(eventName string, attendees []string) {
+	if len(attendees) > 0 {
+		to := config.SurveyMissingEmail
+		subject := "Missing emails for survey: " + eventName
+		bodyText := "The following people did not receive a survey for this protest due to not having a valid email address: "
+		bodyText += strings.Join(attendees, ", ")
+		bodyHtml := bodyText
+		sendEmail(to, subject, bodyText, bodyHtml)
+	}
+}
+
+func sendEmail(to string, subject string, bodyText string, bodyHtml string) {
+	from := config.SurveyFromEmail
+	bodyHtml += `<br /><img src="https://adb.dxe.io/static/img/logo1.png" height="46" width="50">`
+	// EnvConfig uses the AWS credentials in the environment
+	// variables $AWS_ACCESS_KEY_ID nd $AWS_SECRET_KEY.
+	_, err := ses.EnvConfig.SendEmailHTML(from, to, subject, bodyText, bodyHtml)
+	if err != nil {
+		log.Printf("Error sending email: %s\n", err)
+	}
+}
+
+func bulkSendEmails(event model.Event, subject string, bodyText string, bodyHtml string) {
+	for _, recipient := range event.AttendeeEmails {
+		log.Println("Sending email to:", recipient)
+		// Send email
+		sendEmail(recipient, subject, bodyText, bodyHtml)
+	}
+	sendMissingEmail(event.EventName, event.AttendeeMissingEmails)
+}
+
+func updateSurveyStatus(db *sqlx.DB, eventId int) {
+	// Update "survey_sent" to true (1)
+	_, err := model.UpdateEventSurveyStatus(db, model.Event{
+		ID:         eventId,
+		SurveySent: 1,
+	})
+	if err != nil {
+		log.Println("ERROR:", err)
+	}
+}
+
+func surveyMeetup(db *sqlx.DB, queryDate string) {
```

Comment:
These three surveyFoo functions seem very similar. I'd try to write them as a single `surveyEvent` function that parameterizes the different conditions.

Reply:
Was thinking that would be a good idea, but wasn't sure the best way to implement it. Will do some more digging.

Reply:
I think I figured it out.

Reply:
It looks like the main differences are the model.GetEventOptions, the description used for the bodyText/bodyHtml, and the URL. I'd think those should all be easily made into function parameters.

If having a bunch of function parameters becomes too unwieldy, you can put them into a struct type so that field keys (i.e., writing `Foo: x, Bar: y`) makes it clearer what each value means.

Reply:
This is implemented in next commit.